src/coremods/core_channel/cmd_kick.cpp

```diff
@@ -42,9 +42,14 @@ CmdResult CommandKick::Handle (const std::vector<std::string>& parameters, User
 	else
 		u = ServerInstance->FindNick(parameters[1]);
 
-	if ((!u) || (!c) || (u->registered != REG_ALL))
+	if (!c)
 	{
-		user->WriteNumeric(Numerics::NoSuchNick(c ? parameters[1] : parameters[0]));
+		user->WriteNumeric(Numerics::NoSuchChannel(parameters[0]));
+		return CMD_FAILURE;
+	}
+	if ((!u) || (u->registered != REG_ALL))
```

Comment:
This will always result in an error as a target cannot be both a user and a channel.

Reply:
The code that this is replacing is ```if ((!u) || (!c) || (u->registered != REG_ALL))```

Git diffs isn't showing https://github.com/B00mX0r/inspircd/blob/1e6d366930fde3c21f0ebf3485983905ba62491a/src/coremods/core_channel/cmd_kick.cpp#L34,#L43, which is the relevant part that shows how both of them should be defined (you need a valid user to kick and a channel to kick them from).